.github/workflows/checks.yaml

```
@@ -0,0 +1,41 @@
+name: checks
+
+on:
+  workflow_call:
+    inputs:
+      publish:
+        required: false
+        type: string
+        default: "false"
+
+jobs:
+  check-style:
+    runs-on: ubuntu-latest
+    container:
+      image: rapidsai/ci-conda:latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Check style
+        run: "ci/check_style.sh"
+  conda-build:
+    needs: check-style
+    runs-on: ubuntu-latest
+    container:
+      image: rapidsai/ci-conda:latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Python build
+        run: "./ci/build_python.sh ${{ inputs.publish }}"
```

Avoid script injection: https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#example-of-a-script-injection-attack

Also applies elsewhere.
```suggestion
        env:
          PUBLISH: ${{ inputs.publish }}
        run: "./ci/build_python.sh ${PUBLISH}"
```